src/util/device_path.rs

```diff
@@ -0,0 +1,44 @@
+// Copyright 2017 CoreOS, Inc.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use base::Status;
+use protocol::{DevicePathProtocol, DevicePathUtilitiesProtocol};
+use void::CVoid;
+use util::*;
+
+pub fn create_file_device_node(filename: &str) -> Result<&DevicePathProtocol, Status> {
+    str_to_utf16_ptr(filename).and_then(|filename_ptr| {
+        let filename_len = utf16_strlen(filename_ptr);
+        let node_size_bytes = 4 + (filename_len + 1) * 2;
+
+        ::get_system_table()
+            .boot_services()
+            .locate_protocol::<DevicePathUtilitiesProtocol>(0 as *const CVoid)
+            .and_then(|utilities| {
+                utilities.create_device_node(4, 4, node_size_bytes as u16)
```

Comment:
Yes, probably. Realistically the best thing to do is to flesh out all the DevicePath types and subtypes from the spec, I was just a little hesitant to do that since we're not using many of them.